unittests/linuxliketests.py

```diff
@@ -256,6 +256,36 @@ def test_pkgconfig_gen_deps(self):
             content = f.read()
             self.assertNotIn('-lstat2', content)
 
+    def test_pkgconfig_relocatable(self):
```

Comment:
Why is this only in linuxliketests? Do we not want to check that it works on Windows?

Reply:
That's where I found other tests for pkgconfig, so I just assumed meson did not use pkgconfig on windows (I haven't used a windows box in years). Should I move this to common? Or where is a good place for it?

Reply:
As a unittest (rather than a project test) it should likely go in `unittests/allplatformstests.py`.

As for whether you can use pkg-config on Windows, you definitely *can*, and some people do. Meson supports it. There's a culture on Windows of refusing to use pkg-config because Windows users think that as tech that originated in Unix land, pkg-config must have cooties or something.

cmake is actually very popular on Windows for exactly that reason -- it both uses its own in-house format + config parser, and supports relocatable installs (much to the dismay of Linux distro packagers! This relocation support actually causes Linux distro bugs if you invoke /bin/cmake instead of /usr/bin/cmake, and as a result the includedir is "relocated" to /include which obviously does not exist).

So cmake users think that cmake supports relocatable installs and pkg-config doesn't. For that matter, I am not much of a Windows user, so I didn't realize pkg-config supported pcfiledir either -- but...

ironically, pcfiledir was *added* to the pkg-config spec for the sake of Windows support, and this is the reason why I felt a PR for this feature is worth it, and particularly, why it should be a module option rather than used exclusively inside one or two projects that expect to be relocatable. Because I suspect that Windows users will want to set this option as a matter of course. :)

Reply:
> As a unittest (rather than a project test) it should likely go in unittests/allplatformstests.py.

Sounds good, I'll put it there.

> As for whether you can use pkg-config on Windows, you definitely can, and some people do. Meson supports it. There's a culture on Windows of refusing to use pkg-config because Windows users think that as tech that originated in Unix land, pkg-config must have cooties or something.…

Thanks for the explanation, I learned something new today. :+1: Both pkg-config on Windows and cmake. Luckily I haven't encountered the `/bin/cmake` vs `/usr/bin/cmake` but a fair deal of cmake based packages trying to vendor all the deps instead of using the deps already installed. I assume I have Windows to thank for that. :smile: